pkg/runtime/runtime.go

```diff
@@ -104,16 +146,34 @@ func WithPodGroupPolicy(pgPolicy *trainer.PodGroupPolicy) InfoOption {
 	}
 }
 
-func WithPodSpecReplicas(replicaName string, replicas int32, podSpec corev1.PodSpec) InfoOption {
+// DEPRECATED: Replace WithPodSpecReplicas with WithTemplateSpec
+// once we remove TotalResourceRequest.
+
+func WithPodSpecReplicas(
+	replicaName string, count int32, singlePodRequest corev1.ResourceList, podSpecApply *corev1ac.PodSpecApplyConfiguration,
+) InfoOption {
 	return func(o *InfoOptions) {
 		o.podSpecReplicas = append(o.podSpecReplicas, podSpecReplica{
-			name:     replicaName,
-			replicas: replicas,
-			podSpec:  *podSpec.DeepCopy(),
+			name:              replicaName,
+			count:             max(count, 1),
+			podSpecApply:      podSpecApply,
+			singlePodRequests: singlePodRequest,
 		})
 	}
 }
 
+func WithTemplateSpec(objApply any) InfoOption {
```

Comment:
@astefanutti @andreyvelich As we discussed in other threads, I refined these so that we can handle ApplyConfiguration.
However, we keep both `PodSpecReplicas` and `TemplateSpec` in this PR since if we combine these into one in the PR, the `PodRequests` calculation mechanism is broken, and then we need to fix all plugins.